modules/pathseq/1.0/pathseq.smk

```diff
@@ -0,0 +1,328 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Kostiantyn Dreval
+# Module Author:    Kostiantyn Dreval
+# Contributors:     N/A
+
+# Check that the oncopipe dependency is up-to-date. Add all the following lines to any module that uses new features in oncopipe
+min_oncopipe_version="1.0.11"
+import pkg_resources
+try:
+    from packaging import version
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    logger.warning(
+                '\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}'
+                "\n" f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m'
+                )
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section 
+
+
+##### SETUP #####
+
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["pathseq"]`
+CFG = op.setup_module(
+    name = "pathseq",
+    version = "1.0",
+    subdirectories = ["inputs", "flagstat", "pathseq", "outputs"],
+)
+
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _pathseq_input_bam,
+    _pathseq_input_fasta,
+    _pathseq_fasta_dictionary,
+    _pathseq_download_microbe_references,
+    _pathseq_download_taxonomy,
+    _pathseq_calculate_ebv,
+    _pathseq_output,
+    _pathseq_all
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _pathseq_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"]
```

Comment:
Could you please include the bai as an input to this rule, too? It ensures that any alignment modules that might be run upstream complete their index-producing steps.

Reply:
Thanks! I did include the bai index as input and pushed the updates, so this is ready for re-review.